### DISCUSSION TITLE
How to override `_tasks` via API?

### DISCUSSION THREAD

**Discussion by AlexeyShchukinSecurrency** [copier-org/copier#838] [Q&A] [answered]
Hi there,

According to the documentation:

> Can be overridden with the `tasks` API option, but not from CLI.

Assume, I'd like to override the task defined inside a template:
```yaml
---
_tasks:
  - "echo foo"
  - "echo bar"
```

Unfortunately, it seems `Worker` class doesn't have such an option.

```python
@dataclass(config=ConfigDict(extra=Extra.forbid))
class Worker:
    """Copier process state manager.
   ...
    """

    src_path: Optional[str] = None
    dst_path: Path = field(default=Path("."))
    answers_file: Optional[RelativePath] = None
    vcs_ref: OptStr = None
    data: AnyByStrDict = field(default_factory=dict)
    exclude: StrSeq = ()
    use_prereleases: bool = False
    skip_if_exists: StrSeq = ()
    cleanup_on_error: bool = True
    defaults: bool = False
    user_defaults: AnyByStrDict = field(default_factory=dict)
    overwrite: bool = False
    pretend: bool = False
    quiet: bool = False
```

Is there any other way to achieve override template tasks?

Thanks in advance.

**Comment by yajo** [comment, marked as answer]
Ah no sorry, tasks are meant to be defined in the template. 

I have plans in #240 to improve that part, as tasks could contain malicious code. For example, the user could choose not to run them and, instead, run some scripts after an update manually. 

For now it's a docs bug. Fixed in 8cb181aa5edd738f1fbf264e353f1419da7af794.

**Reply by AlexeyShchukinSecurrency** [reply]
Got it. Thanks for your reply!
